test/tftp/tftp_api_test.rb

```diff
@@ -26,22 +26,64 @@ def test_api_can_fetch_boot_file
   end
 
   def test_api_can_create_syslinux_tftp_reservation
```

Comment:
I have broken into two suites - one is testing the API, the other one is testing the Server and its implementations. Both paths and code logic are executed. I am keeping stubbing because when I started with tempfiles solution, it actually was a longer code. TFTP module does not change many often.